patterns/1-initial/transitioning-contractor-code-to-innersource-model.md

```diff
@@ -0,0 +1,75 @@
+## Title
+
+Transitioning Contractor Code to InnerSource Model
+
+## Patlet
+
+Contract developers are often not motivated to engage in InnerSource activities, which may be beyond the scope of their contract. This patlet describes how you can focus on transitioning the contractor project after the fact to an InnerSource project.
+
+## Problem
+
+Contractor developers are often not motivated (through forces described below) to not engage in InnerSource activities. Once delivered, and even if the code is made visible, their projects are often less likely to be part of successful InnerSourced engagements.
+
+## Context
+
+This problem exists where an organization either:
+
+- out-sources the development of a well defined project or
+- engagages external firms for staff augmentation and has mixed teams of permanent employees with a large percentage of contract staff. This issue may occur less often in a mixed team where InnerSource principles and ways of working are already established within the organization, but the forces described below are still often at play.
+
+## Forces
+
+Contractor Motivation and Constraints:
+
+- Often contracts with third-party developers are very focused on delivering an end result in the fastest possible fashion. As a result, all InnerSource activities (e.g. responding to third-party PRs) are considered to be distractions or something that will “slow down” ultimate delivery.
+- There is also often a concern that accepting code from other parts of the business might introduce security risks, scope creep or other issues that would subsequently have to be resolved by the contract team.
+- Above and beyond the idea that InnerSource may slow down the project, there is often an additional concern that accepting PRs from other parts of the company may “muddy the waters” when it comes to assessing what parts of the project were completed/delivered by the contracted developers.
+
+All of the above can mean that even if an individual contract developer wants to engage in InnerSource, there are system-level constraints pushing them not to.
+
+It should be noted that the above scenario is indirectly impacted by:
+
+- The norms around defining Statements of Work for third party contractors
+- Pressures to reduce contractor costs during procurement
+- Ability to tie contributions to payment at a granular level.
+
+It could also be noted that the Contractor motivations in this instance is almost like a more extreme instance of the oft-reported organizational/budgetary constraints that might exist for some internal business units. (Not sure if this is relevant, but it does seem to be an extreme case of what is reported as common objections even in internal teams).
+
+## Solutions
+
+At the outset of the project, clearly define:
+
+- That the code will be made visible by default.
+- How the project will be transitioned to an InnerSource project. (Zack can add more explicit details here)
```

Comment:
```suggestion
- How the project will be transitioned to an InnerSource project. This could be similar to a transition of ownership plan for an open source project which should include: An identified team of new maintainers, documentation on how to responsibilities/build/test/deploy/release, and an announcement to stakeholders regarding the transition.
- That the architecture of the code should be modular and ready for component reuse.
```

Reply:
@zkoppert for the "transition of ownership plan for an open source project", is there a resource that we could link to that explains how this is done in open source?

Reply:
Further I am wondering how typical contracts between company and contractor describe the expected handover behavior (even bevor this pattern is in place)?

As any source code needs to be maintained, the project that a contractor build would need a maintainer afterwards, even if not working in an InnerSource model. How would that typically work?

Reply:
I've seen contract language that specifies transition deliverables that include: 
- Written documentation describing functionality, architecture, and common processes like releasing, patching, deploying, etc.
- A prescribed number of hours on the contract set aside for the contractor to meet with the identified long term maintainer (normally from the company who hired the contractor) for an overview of responsibilities and Q & A.
I haven't seen but it could be tried:
- Apply the 30 day warranty pattern and so the contractor provides 30 days of direct transition support to the new maintainer

Reply:
I don't have any specific resources but I am checking with some other folks to see if they know of any related to transition of ownership.

Reply:
Here is a resource on transitioning leadership in an OSS project that seems relevant: https://contribute.cncf.io/maintainers/governance/leadership-selection/#transition-from-existing-leadership-to-new-governance

Reply:
> * Written documentation describing functionality, architecture, and common processes like releasing, patching, deploying, etc.
> * A prescribed number of hours on the contract set aside for the contractor to meet with the identified long term maintainer (normally from the company who hired the contractor) for an overview of responsibilities and Q & A.
>   I haven't seen but it could be tried:
> * Apply the 30 day warranty pattern and so the contractor provides 30 days of direct transition support to the new maintainer

Would be great work these bullets directly into the Solution.

Reply:
I worked those into my suggestion above.